rdflib/plugins/sparql/operators.py

```diff
@@ -521,8 +521,13 @@ def Builtin_TZ(e: Expr, ctx) -> Literal:
     if not d.tzinfo:
         return Literal("")
     n = d.tzinfo.tzname(d)
-    if n == "UTC":
+    if n is None:
+        n = ""
+    elif n == "UTC":
         n = "Z"
+    elif n.startswith("UTC"):
+        # Replace tzname like "UTC-05:00" with simply "-05:00" to match Jena tz fn
+        n = n[3:]
```

Comment:
This was a required change because previous implementation used tzinfo instances (timezone instances) from `isodate`, but the new implementation always uses timezones from the stdlib. The SPARQL builtin `TZ()` will return the "timezone name" if known, in python this uses the `tzname()` function.
The difference is that stdlib timezones and isodate timezones have different `tzname()` generation pattern. In `isodate` an unnamed timezone with "-5H" offset will be have tzname of "-05:00", but in stdlib an unnamed timezone with "-5H" offset will have tzname of "UTC-05:00" (more correct IMHO).

So for consistence and backwards compatibility, this was change was added to normalize the tzname output, I tested Jena's `TZ()` SPARQL function too, and it also outputs "-05:00" like `isodate` did.